Paper JOSS (#43)
YoungFaithful/CapacityExpansion.jl: paper/paper.md

    affiliation: 1
  - name: Holger Teichgraeber
    orcid: 0000-0002-4061-2226
    affiliation: 1
  - name: Adam R. Brandt
    orcid: 0000-0002-2528-1473
    affiliation: 1
affiliations:
 - name: Department of Energy Resources Engineering, Stanford University
   index: 1
date: 4 October 2019
bibliography: paper.bib
---

# Summary

``CapacityExpansion`` is a Julia implementation of an input-data-scaling capacity expansion modeling framework [@Bezanson:2017]. It provides an extensible, multi-carrier, simple-to-use generation and transmission capacity expansion model that allows to address a diverse set of research questions in the area of energy systems planning and can be used to plan and validate energy systems at scales ranging from quarters to entire global regions.
``CapacityExpansion`` provides simple integration of (clustered) time-series, geographical, cost, and technology input data. The software features a modular model setup and an investment and dispatch optimization that uses the JUMP modeling language [@Dunning:2017]. An interface is provided between the optimization result and further analysis.

# Infrastructure planning in the energy sector

> Energy systems convert different energy resources to meet desired demands like electric and thermal energy demands. Political, economical, and technological changes require changing the infrastructure of energy systems. Changing the infrastructure has to balance multiple political, environmental, and economical objectives. Capacity expansion planning can be an important tool during the planning process [@Gacitua:2018].

Reviewer: ```suggestion
Energy systems convert different energy resources to meet desired demands like electric and thermal energy demands. Political, economical, and technological changes require expansions of the infrastructure of energy systems. Expanding the infrastructure has to balance multiple political, environmental, and economical objectives. Capacity expansion planning can be an important tool during the planning process [@Gacitua:2018].
```

Author: I see the point. ;)

> Capacity expansion planning is used to compute cost-optimal energy system designs under given sets of constraints. The resulting cost-optimal energy system design can be used to inform policy decisions that incentivize the industry to invest in this design [@Johnston:2013]. Similarly, cost-optimal energy system designs can be used by companies for their investment strategies.

Reviewer: ```suggestion
Capacity expansion planning is used to compute cost-optimal energy system designs under given sets of constraints from the perspective of a central planner. The resulting cost-optimal energy system design can be used to inform policy decisions that incentivize the industry to invest in this design [@Johnston:2013]. Similarly, cost-optimal energy system designs can be used by companies for their investment strategies.
```

Author: Great addition

Aspects of the energy system design that capacity expansion planning aims to answer are what the optimal technology mix is in regards of location, time, and installed generation, conversion, storage, and transmission capacities. The design optimization is done while using an integrated dispatch formulation to ensure that supply can equal demand at all nodes and time steps. The model determines the costs, emissions, power generation, energy storage, and power flows based on the installed capacities.

> Capacity expansion planning is formulated as mathematical optimization problem. Like any optimization problem, capacity expansion planning has certain degrees of freedom, consists of constraints, and consists of an objective function that is minimized: Typical degrees of freedom (sometimes called decision variables) are installed capacities, power generation, energy storage, and power flows. We divide constraints into internal and external constraints: The internal constraints ensure that the model is physically consistent in itself and following the rules of thermodynamics, e.g. energy balances ensure energy conservation over time. External constraints restrict the solution space to external conditions like costs, demands, available energy resources as well as other political, environmental, economical, or technological constraints. The objective function to determine cost-optimal investment can be total system cost, the net present value, or another business-oriented cost measure.

Reviewer: ```suggestion
Capacity expansion planning is formulated as a mathematical optimization problem. Like any optimization problem, capacity expansion planning has certain degrees of freedom, consists of constraints, and consists of an objective function that is minimized: Typical degrees of freedom (also called decision variables) are installed capacities, power generation, energy storage, and power flows. We divide constraints into internal and external constraints: The internal constraints ensure that the model is physically consistent in itself and following the rules of thermodynamics, e.g. energy balances ensure energy conservation over time. External constraints restrict the solution space to external conditions like costs, demands, available energy resources as well as other political, environmental, economical, or technological constraints. The objective function to determine cost-optimal investment can be total system cost, the net present value, or another business-oriented cost measure.
```

Reviewer: Is the distinction between internal and external necessary? I find it slightly confusing, maybe take the distinction out and just list constraints.

Author: I agree, that it is not necessary.

Dispatch planning formulations are very similar to capacity expansion formulations. However, the installed capacity is no degree of freedom, but introduced as an external constraint.

# Package features

> The model class of ``CapacityExpansion`` is capacity expansion planning and generation and transmission capacity expansion planning are combined. The model is setup as a linear optimization model that models energy systems based on the provided input data. Multiple energy carriers can be modeled, which makes the software suitable for research of sector coupling technologies. Technologies can be defined that belong to dispathable or non-dispathable generation, conversion, storage, transmission, or demand. The decision variables of the model are investment and dispatch and the total system costs are minimized.

Reviewer: Something is missing in the first sentence, gramatically.

Author: Would:
`The model class of ``CapacityExpansion`` is capacity expansion planning and it combines generation and transmission capacity expansion planning. ` make more sense?

The following key features are provided by ``CapacityExpansion``. The usage and mathematical formulation is explained in detail within the software's documentation.

> - *Modeling language*: The implementation in Julia enables the usage of a high-level programming language, and short times for data handling at the same time. The implementation in Julia also allows the usage of the mathematical optimization language JUMP, which enables the optimization formulation close to mathematical writing. The integration into the Julia ecosystem further allows to handle the process from data preparation to figure export within one programming language and software.

Reviewer: ```suggestion
- *Modeling language*: The implementation in Julia enables the usage of a high-level programming language, and short times for data handling at the same time. The implementation in Julia also allows the usage of the mathematical optimization language JUMP [CITE JUMP], which enables the optimization formulation close to mathematical writing. The integration into the Julia ecosystem further allows to handle the process from data preparation to figure export within one programming language and software.
```

Author: Added

- *Provided input data*: The input data for single and multi-node representations of electricity systems are provided for California (USA) and Germany (EU). The input data provides hourly time-series input data for multiple years, aggregated geographic information about existing generation and transmission, cost information including the monetary and life cycle assessment costs, and necessary technology data. This input data can be used to address many research questions.

- *The generalized import of input data*: Modeling other energy systems is possible by adjusting the input data. The package extracts all information needed to model a specific energy system based on a few standardized input files. The time-series, geographic, and cost data can be edited as tables and integrated using the `.csv`-file format. The technology input data can be edited like a tree structure and integrated using the `.yml`-file format.

> - *Integration of ``TimeSeriesClustering``*: Time-series data like the demand, available solar factors, and available wind factors are used as an input to model the temporal variance of the energy system. Aggregating the time-series input data is commonly done to reduce the computational complexity of the optimization. ``CapacityExpansion`` is well integrated with the Julia package ``TimeSeriesClustering``. This integration allows using the typical time-series aggregation methods, integrated testing of the temporal resolution, and an integrated feedback loops between the optimization result and time-series aggregation.

Reviewer: ```suggestion
- *Integration of ``TimeSeriesClustering``*: Time-series data like the demand, available solar factors, and available wind factors are used as an input to model the temporal variance of the energy system. Aggregating the time-series input data is commonly done to reduce the computational complexity of the optimization. ``CapacityExpansion`` is well integrated with the Julia package ``TimeSeriesClustering`` [CITE JOSS PAPER TIMESERIESCLUSTERING]. This integration allows using the typical time-series aggregation methods, integrated testing of the temporal resolution, and an integrated feedback loops between the optimization result and time-series aggregation.
```

Reviewer: Also, please add citation to Applied Energy clustering paper after `typical time-series aggregation methods`.

Author: Added the citations.

- *Seasonal storage*: A recent seasonal storage formulation from Kotzur is implemented to allow time-series aggregation and modeling of seasonal storage at the same time [@Kotzur:2018].

- *Modular model setup*: The model setup is flexible and based on a modular setup. Depending on the configuration different technology groups can be activated or deactivated, a green or brown field study performed, emissions limits can be enforced, different storage models can be used, and an integrated investment and dispatch or pure dispatch optimization can be run.

# ``CapacityExpansion`` within the broader ecosystem

> ``CapacityExpansion`` is the first package to provide capacity expansion planning in Julia [@Bezanson:2017].

Reviewer: I would take out the reference to Bezanson here and add it further up. Bezanson doesnt talk about CEP, so may be confusing

Author: I already had one reference at the beginning of the text and I'll take this one out.
